src/puppetlabs/pcp/client.clj

```diff
@@ -111,7 +111,9 @@
     (let [sessions (.getOpenSessions websocket-client)]
       (log/debug (i18n/trs "Sending WebSocket ping"))
       (doseq [session sessions]
-        (.. session (getRemote) (sendPing (ByteBuffer/allocate 1))))))
+        ;; sessions in the CLOSING state are in OpenSessions
+        (when (.isOpen session)
+          (.. session (getRemote) (sendPing (ByteBuffer/allocate 1)))))))
```

Comment:
Yup, this at least shortens the race and the outcome of losing it is just a superfluous log message. The exception is pretty generic and I was concerned that handling it would squash errors other than "closing".